x-pack/plugins/reporting/server/deprecations/reporting_role.ts

```diff
@@ -148,7 +150,9 @@ async function getRoleMappingsDeprecations(
   deprecatedRoles: string[],
   docLinks: DocLinksServiceSetup
 ): Promise<DeprecationsDetails[]> {
-  const usingDeprecatedConfig = !reportingCore.getContract().usesUiCapabilities();
+  const usingDeprecatedConfig =
+    !reportingCore.getContract().usesUiCapabilities() &&
+    reportingCore.getConfig().statefulSettings.enabled;
```

Comment:
Same in this file as above - I think we no longer need this change since the deprecations internally will be looking at the right values in the config